importer.go

```diff
@@ -122,28 +122,56 @@ func (i *Importer) readViaCache(imp string) (jsonnet.Contents, error) {
 		return content, nil
 	}
 
-	u, err := addScheme(imp)
+	content, err := i.fetch(imp)
 	if err != nil {
 		return noContent, err
 	}
-	content, err := fetch(u, i.fetcher())
-	if err == nil {
-		i.cache[imp] = content
+
+	i.cache[imp] = content
+	return content, nil
+}
+
+func (i *Importer) fetch(imp string) (jsonnet.Contents, error) {
+	r, err := i.open(imp)
+	if r == nil || err != nil {
+		return noContent, err
 	}
 
-	return content, err
+	defer r.Close()
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return noContent, err
+	}
+
+	return jsonnet.MakeContents(string(b)), nil
 }
 
-func (i *Importer) fetcher() URLFetcher {
-	if i.Fetcher == nil {
-		// Create a http.Client with a transport handling file:// urls
-		// https://golang.org/pkg/net/http/#NewFileTransport
-		t := &http.Transport{}
-		t.RegisterProtocol("file", http.NewFileTransport(http.Dir("/")))
+func (i *Importer) open(imp string) (io.ReadCloser, error) {
+	if !isNetpath(imp) {
+		r, err := os.Open(imp)
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return r, err
+	}
+
+	resp, err := i.fetcher().Get("https:" + imp)
+	if err != nil {
+		return nil, err
+	}
 
-		i.Fetcher = &http.Client{Transport: t}
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, nil
+	} else if resp.StatusCode != http.StatusOK {
```

Comment:
what about other 2xx and 3xx status codes?

Reply:
The standard `http.Client` handles redirects (3xx). I don't think the other 2xx status codes mean "here's what you wanted", so I treat them as errors too.

Reply:
👍